joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/StaticCallLinker.scala

```diff
@@ -1,64 +1,39 @@
 package io.joern.x2cpg.passes.callgraph
 
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.nodes.*
 import io.shiftleft.codepropertygraph.generated.{DispatchTypes, EdgeTypes}
-import io.shiftleft.passes.CpgPass
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.passes.{ConcurrentWriterCpgPass, ForkJoinParallelCpgPass}
+import io.shiftleft.semanticcpg.language.*
 import org.slf4j.{Logger, LoggerFactory}
 
-import scala.collection.mutable
+class StaticCallLinker(cpg: Cpg) extends ForkJoinParallelCpgPass[Seq[Call]](cpg) {
 
-class StaticCallLinker(cpg: Cpg) extends CpgPass(cpg) {
-
-  import StaticCallLinker._
-  private val methodFullNameToNode = mutable.Map.empty[String, List[Method]]
-
-  override def run(dstGraph: DiffGraphBuilder): Unit = {
+  private val logger: Logger = LoggerFactory.getLogger(classOf[StaticCallLinker])
+  private val BATCH_SIZE     = 100
 
-    cpg.method.foreach { method =>
-      methodFullNameToNode.updateWith(method.fullName) {
-        case Some(l) => Some(method :: l)
-        case None    => Some(List(method))
-      }
-    }
+  override def generateParts(): Array[Seq[Call]] = {
+    cpg.call.grouped(BATCH_SIZE).toArray
+  }
 
-    cpg.call.foreach { call =>
+  override def runOnPart(builder: DiffGraphBuilder, calls: Seq[Call]): Unit = {
+    calls.foreach { call =>
       try {
-        linkCall(call, dstGraph)
+        call.dispatchType match {
+          case DispatchTypes.STATIC_DISPATCH | DispatchTypes.INLINED =>
+            val resolvedMethodOption = cpg.method.fullNameExact(call.methodFullName)
+            resolvedMethodOption.foreach(dst => builder.addEdge(call, dst, EdgeTypes.CALL))
+            val size = cpg.method.fullNameExact(call.methodFullName).size
+            // Add the debug logs with number of METHOD nodes found for given method full name
+            if size > 1 then logger.debug(s"Total ${size} METHOD nodes found for -> ${call.methodFullName}")
+          case DispatchTypes.DYNAMIC_DISPATCH =>
+          // Do nothing
+          case _ => logger.warn(s"Unknown dispatch type on dynamic CALL ${call.code}")
+        }
       } catch {
         case exception: Exception =>
           throw new RuntimeException(exception)
```

Comment:
We should probably not throw an exception here. I know this is existing code, but could you replace this with an error log?